plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go

```diff
@@ -26,11 +26,30 @@ func ClusterRoles() []rbacapi.ClusterRole {
 	return []rbacapi.ClusterRole{
 		// TODO update the expression of these rules to match openshift for ease of inspection
 		{
+			// a "root" role which can do absolutely anything
 			ObjectMeta: api.ObjectMeta{Name: "cluster-admin"},
 			Rules: []rbacapi.PolicyRule{
 				{Verbs: []string{"*"}, APIGroups: []string{"*"}, Resources: []string{"*"}},
 				{Verbs: []string{"*"}, NonResourceURLs: []string{"*"}},
 			},
 		},
+
+		{
+			// a role which provides just enough power to discovery API versions for negotiation
+			ObjectMeta: api.ObjectMeta{Name: "system:discovery"},
```

Comment:
Why do you use the `system:` prefix?  It is not like usernames or groupnames, where there is no collision control.  For objects like ClusterRoles, the apiserver provides collision control based on names.